templates/al2023/runtime/gpu/nvidia-kmod-load.sh

```diff
@@ -42,9 +47,23 @@ function devices-support-open() {
   return 0
 }
 
+function device-supports-grid() {
+  for NVIDIA_GRID_SUBDEVICE in "${NVIDIA_GRID_SUBDEVICES[@]}"; do
+    for NVIDIA_SUBDEVICE in $(lspci -n -mm -d 10de:  | tr -d '"' | awk '{print $4":"$7}'); do
```

Comment:
consistency with the existing method: https://github.com/awslabs/amazon-eks-ami/blob/fd505f6c036c8ed1729a5cd779ff4c89c980c32d/templates/al2023/runtime/gpu/nvidia-kmod-load.sh#L41

```suggestion
    for NVIDIA_SUBDEVICE in $(lspci -n -mm -d "${NVIDIA_VENDOR_ID}:" | awk '{print $4":"$7}' | tr -d '"'); do
```